### PR TITLE
Travis opt

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Lets see what happens if we move some things out of our travis configuration.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,28 +5,13 @@ rust:
...
   export PATH=$HOME/.local/bin:$PATH
 script:
 - |
-  export RUSTFLAGS="-C codegen-units=2"
-  if [[ "$TEST_SUITE" == "binaries" ]]; then
-    make bin-test
-  else # $TEST_SUITE == "libraries"
-    make lib-test
-  fi
+  make
```
Comment (mario-kr) [inline .travis.yml]: this doesnt call any tests afaik. you probably want them, though you might as well
```sh
make
make bin-test
make lib-test
```
build time for the last 2 should be near to 0.

btw, does calling `make bin-test` even build imag-bin? It should, but does it?

**Comment by mario-kr** [comment]
[this](https://github.com/rust-lang/rust/blob/master/RELEASES.md#cargo) should explain our troubles with beta 😄

File: .travis.yml (hunk 1 above)
Comment (matthiasbeyer) [inline .travis.yml]: Good point(s)!

> btw, does calling `make bin-test` even build imag-bin? It should, but does it?

It should, yes.

Comment (matthiasbeyer) [inline .travis.yml]: Nah, thinking about it again - it shouldn't. It should do `make imag-bin-test`, but there is no such thing, so ... everything is fine!

**Comment by mario-kr** [comment]
it's kinda infuriating that you dont even `cargo check` before pushing stuff to travis.
btw, the reason it didnt fail on the `fix imag bin build script` branch is that `make bin-test` doesnt build `imag`, which you said was wanted (no `imag-bin-test` target is specified was your point I think)

**Comment by matthiasbeyer** [comment]
> it's kinda infuriating that you dont even cargo check before pushing stuff to travis.

If someone else can compute things for me, why should I do it? :smile:

**Comment by matthiasbeyer** [comment]
@mario-kr can I have an ACK or NACK?

**Comment by mario-kr** [comment]
ACK

**Pull request merged**